src/components/app.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import {compose} from 'redux';
+import {connect} from 'react-redux';
+import {Route, withRouter} from 'react-router-dom';
+
+import HeaderBar from './header-bar';
+import LandingPage from './landing-page';
+import Dashboard from './dashboard';
+import RegistrationPage from './registration-page';
+import {refreshAuthToken} from '../actions/auth';
+
+export class App extends React.Component {
+    componentDidMount() {
+        if (this.props.hasAuthToken) {
+            // Try to get a fresh auth token if we had an existing one in
+            // localStorage
+            this.props.dispatch(refreshAuthToken());
```

Comment:
Something we'll want to explain and something I'm curious about: business logic of refreshing token each time the app loads.